python/pyspark/sql/column.py

```diff
@@ -296,12 +299,17 @@ def getItem(self, key):
         +----+------+
         |   1| value|
         +----+------+
-
-        .. versionchanged:: 3.0
-           If `key` is a `Column` object, the indexing operator should be used instead.
-           For example, `map_col.getItem(col('id'))` should be replaced with `map_col[col('id')]`.
         """
-        return _bin_op("getItem")(self, key)
+        if isinstance(key, Column) and not is_instance_of(
+                SparkContext._gateway,
+                key._jc.expr(),
+                "org.apache.spark.sql.catalyst.expressions.Literal"):
```

Comment:
So we allow Column of Literal?

Reply:
That was my current take. It was because `getField` support a literal only at this moment as well (whereas Scala side only support `String`). I took a look for `functions.py`, and seems it's more consistent to just let it take string only, and no `Column` instance. Thanks for pointing this one ..

I will update this PR accordingly later tomorrow.